src/components/common/BackLink.tsx

```diff
@@ -2,12 +2,18 @@ import React from 'react';
 import Link from '@material-ui/core/Link';
 import { NavLink } from 'react-router-dom';
 
-const BackLink = ({ to }: { to: string }) => {
-  return (
-    <Link color="secondary" variant="body2" component={NavLink} to={to}>
-      &#60; Back
-    </Link>
-  );
-};
+const BackLink = ({ to }: { to: string }) => (
+  <Link color="secondary" variant="body2" component={NavLink} to={to}>
+    &#60; Back
+  </Link>
+);
+
+export const BackButton = ({ onClick }: { onClick: () => void }) => (
```

Comment:
So `BackButton` actually uses the `button` component meaning it can accept an `onClick`. Whereas `BackLink` will not accept an `onClick` but only accepts a link to navigate to (using react-router). I've kept them separate because I wasn't ready to affect code that I didn't have to touch during this cleanup. 

I see these options if we want to have one ONLY:
1. Have all the cases where it's a `BackLink` be turned into a `BackButton` and pass `onClick` prop which navigates to the link (cleaner)
    - e.g. `onClick = () => { history.push(LINK) }` is passed as a prop
2. Have all the cases where it's a `BackLink` be turned into a `BackButton` and pass a different optional prop (e.g. `to`) that will build the `onClick` inside the `BackButton` component
    - e.g. `to = '/events'` is passed as a prop and inside the component it's passed to, we create `onClick = () => { history.push(to) }`
    - I have some reservations about this since BOTH the `onClick` and `to` prop will be optional where we really need to have exactly ONE of them so not super clean but i think it is enforceable via typescript ([see here](https://stackoverflow.com/questions/40510611/typescript-interface-require-one-of-two-properties-to-exist))
3. Same as option 2 but instead of building an `onClick` for the `to` prop case, we do something like the following:
(Ignore the red underline due to linter error because the single line of code is too long)
![image](https://user-images.githubusercontent.com/22457859/93345000-54865680-f800-11ea-8f86-f9da1e06d7ea.png)
Thoughts? @petorsong

Reply:
thanks for the detailed overview!

i did have option (2) in mind, but i can see how that's kinda hacky if we require one of two optional props - there are a bunch of usecases for both `BackLink` and `BackButton` right now so it's kinda messy to handle in this PR

i've created a new ticket in our backlog to address this separately, so we won't worry about it here
https://www.notion.so/uwblueprintexecs/back-link-button-cleanup-6b24e137ca1648368f0c1f326485142f

Reply:
Sounds good, thanks!